### PR TITLE
Lwdev 5690 enroll dust cashin to the me

### PR DESCRIPTION
**Pull request by PaterSantyago**

### CODE REVIEW

**Comment by tarurar** [comment]
Please resolve conflicts before merging.

File: src/Lykke.Job.BlockchainCashinDetector.Core/Domain/CashinAggregate.cs [hunk 1]
```diff
@@ -201,9 +217,13 @@ public bool OnClientOperationStartRegistered()
...
-            if (!SwitchState(
-                new[] {CashinState.ClientOperationStartIsRegistered, CashinState.EnrolledToMatchingEngine},
-                CashinState.OperationIsFinished))
+            var expectedStates = new[]
+            {
+                CashinState.ClientOperationStartIsRegistered,
+                CashinState.EnrolledBalanceIncreased
+            };
+
+            if (!SwitchState(expectedStates, CashinState.EnrolledBalanceReset))
```
Comment (KonstantinRyazantsev) [inline src/Lykke.Job.BlockchainCashinDetector.Core/Domain/CashinAggregate.cs]: It looks strange, that on operation completed, aggregate switches to the EnrolledBalanceReset state

File: src/Lykke.Job.BlockchainCashinDetector.Core/Domain/CashinAggregate.cs [hunk 2]
```diff
@@ -214,6 +234,18 @@ public bool OnOperationCompleted(string transactionHash, long transactionBlock,
             TransactionAmount = transactionAmount;
             TransactionBlock = transactionBlock;
             Fee = fee;
+            
+            return true;
+        }
+
+        public bool OnEnrolledBalanceReset()
+        {
+            if (!SwitchState(CashinState.EnrolledBalanceReset, CashinState.OperationIsFinished))
```
Comment (KonstantinRyazantsev) [inline src/Lykke.Job.BlockchainCashinDetector.Core/Domain/CashinAggregate.cs]: And this state transition looks strange

File: src/Lykke.Job.BlockchainCashinDetector.Core/Domain/CashinAggregate.cs [hunk 3]
```diff
@@ -24,6 +26,7 @@ public class CashinAggregate
         public string BlockchainAssetId { get; }
         public decimal Amount { get; }
         public string AssetId { get; }
+        public decimal OperationAmount { get; }
```
Comment (KonstantinRyazantsev) [inline src/Lykke.Job.BlockchainCashinDetector.Core/Domain/CashinAggregate.cs]: As we have already discussed, let's rename Amount to the TransactionAmount, and TransactionAmount to the ActualTransactionAmount

**Pull request merged**